src/framework/preferences/js/SeparatedPanelPrefsEditor.js

```diff
@@ -330,7 +339,84 @@ var fluid_2_0_0 = fluid_2_0_0 || {};
     fluid.prefs.separatedPanel.showPanel = function (panel, callback) {
         // A bizarre race condition has emerged under FF where the iframe held within the panel does not
         // react synchronously to being shown
-        setTimeout(callback, 1);
+        fluid.invokeLater(callback);
+    };
+
+    fluid.defaults("fluid.prefs.separatedPanel.lazyLoad", {
+        events: {
+            onLazyLoad: null,
+            onPrefsEditorMessagesPreloaded: null,
+            onCreateSlidingPanelReady: {
+                events: {
+                    onPrefsEditorMessagesLoaded: "onPrefsEditorMessagesPreloaded"
+                }
+            },
+            templatesAndIframeReady: {
+                events: {
+                    onLazyLoad: "onLazyLoad"
+                }
+            }
+        },
+        components: {
+            templateLoader: {
+                createOnEvent: "onLazyLoad"
+            },
+            messageLoader: {
+                options: {
+                    events: {
+                        onResourcesPreloaded: "{separatedPanel}.events.onPrefsEditorMessagesPreloaded"
+                    },
+                    preloadResources: "prefsEditor",
+                    listeners: {
+                        "onCreate.loadResources": {
+                            listener: "fluid.prefs.separatedPanel.lazyLoad.preloadResources",
+                            args: ["{that}", {expander: {func: "{that}.resolveResources"}}, "{that}.options.preloadResources"]
+                        },
+                        "{separatedPanel}.events.onLazyLoad": {
+                            listener: "fluid.resourceLoader.loadResources",
+                            args: ["{messageLoader}", {expander: {func: "{messageLoader}.resolveResources"}}]
+                        }
+                    }
+                }
+            },
+            slidingPanel: {
+                options: {
+                    invokers: {
+                        operateShow: {
+                            funcName: "fluid.prefs.separatedPanel.lazyLoad.showPanel",
+                            args: ["{separatedPanel}", "{that}.events.afterPanelShow.fire"]
+                        }
+                    }
+                }
+            }
+        }
+    });
+
+    fluid.prefs.separatedPanel.lazyLoad.showPanel = function (separatedPanel, callback) {
+        if (separatedPanel.prefsEditor) {
```

Comment:
The purpose is to handle the lazy loading. If the `separatedPanel.prefsEditor` has been instantiated already, we merely just trigger the typical opening flow to show the panel. If it hasn't be instantiated yet, we go through the process of instantiating it. Not sure of how to do this completely declaratively, but open to suggestions.